enumerables.rb

```diff
@@ -0,0 +1,286 @@
+      my_each do |m|
+        if m.class.superclass.name != 'Numeric'
+          classflag = false
+        elsif empty?
+          classflag = false
+        end
+      end
+      classflag
+    elsif !block_given? and type.nil?
+      my_each do |m|
+        return false if m.nil?
+      end
+      true
+    end
+  end
+
+  # my_any? Method
+
+  def my_any?(cond = nil)
+    classflag = false
+    if !cond.nil?
+      if cond.is_a?(Class)
+        my_each do |e|
+          if e.is_a?(cond)
+            classflag = true
+            break
+          end
+        end
+      elsif cond == Numeric
+        my_each do |e|
+          if e.is_a?(Numeric)
+            classflag = true
+            break
+          end
+        end
+      elsif !cond.is_a?(Numeric) and cond.include?(true || false)
+        my_each do |e|
+          if e == cond
+            classflag = true
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = true
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = true
+            break
+          end
+        end
+      end
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = true
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if empty?
+              false
+            else
+              true
+            end
+    else
+      false
+    end
+    classflag
+  end
+
+  # my_none Method
+  def my_none?(cond = nil)
+    classflag = true
+    if !cond.nil?
+      if cond.is_a?(Class)
+        my_each do |e|
+          if e.is_a?(cond)
+            classflag = false
+            break
+          end
+        end
+      elsif cond == Numeric
+        my_each do |e|
+          if e.is_a?(Numeric)
+            classflag = false
+            break
+          end
+        end
+      elsif cond.is_a?(Boolean)
+        my_each do |e|
+          if e == cond
+            classflag = false
+            break
+          end
+        end
+      elsif cond != Class and cond.is_a?(Regexp)
+        my_each do |e|
+          if e.is_a?(String)
+            if e.match?(cond)
+              classflag = false
+              break
+            end
+          end
+        end
+      elsif cond.is_a?(Numeric) || cond.is_a?(String)
+        my_each do |e|
+          if cond == e
+            classflag = false
+            break
+          end
+        end
+      end
+
+    elsif block_given?
+      my_each do |e|
+        if yield(e)
+          classflag = false
+          break
+        end
+      end
+    elsif cond.nil? and !block_given?
+      classflag = if to_a.empty?
+        true
+      else
+        false
+      end
+    else
+      true
+    end
+    classflag
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

#### my_none?
> Remember to make it work with ranges and hashes
- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false].my_none? == [nil, false].none? => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/95547685-bfbae700-09c0-11eb-9642-e4c676e87f40.png)
- Real none?
![image](https://user-images.githubusercontent.com/52209067/95547702-cd706c80-09c0-11eb-8833-525adb3d6417.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- my_none? is retrieving a NameError
![image](https://user-images.githubusercontent.com/52209067/95283704-a461a780-0819-11eb-8edf-4533952f4ba6.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
```ruby
[3,4,7,11].my_none?('3') == [3,4,7,11].none?('3') => true
```
- my_none? retrieves a NameError
![image](https://user-images.githubusercontent.com/52209067/95548907-525c8580-09c3-11eb-968d-33f27df36b04.png)
- Real none?
![image](https://user-images.githubusercontent.com/52209067/95548943-6902dc80-09c3-11eb-916e-77b808d720d7.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?

Reply:
To make this work we have to use block.call(element) instead of yield(element) in my_any?

-> Thanks!!